dask/dataframe/io/parquet.py

```diff
@@ -121,7 +117,8 @@ def read_parquet(path, columns=None, filters=None, categories=None, index=None,
                         columns=[c for c in pf.columns if c in dtypes])
 
     for cat in categories:
```

Comment:
Ah, ok. I assume the same for if there are extra columns specified?

For my own understanding, how expensive would it be to infer the encoding/columns/dtypes from the parquet file? I assume somewhere in the file there's an indication that a column is dictionary encoded?